notebooks/Get Species data from multiple sorces (spocc).qmd

```diff
@@ -0,0 +1,267 @@
+
+![](https://raw.githubusercontent.com/EcoCommons-Australia-2024-2026/ec-notebook_site/main/images/EC_breaker_nobackgoundcolor.png)
+
+# Set-up: R Environment and Packages
+
+Before starting, ensure your environment is correctly set up. This process may take some time as several packages need to be installed.
+
+## S.1 Set the working directory and create a folder for data.
+
+Save the Quarto Markdown file (`.qmd`) to a folder of your choice, and then set the path to your folder as your working directory.
+
+```{r}
+# Set the workspace to the current working directory
+# Uncomment and replace the path below with your own working directory if needed:
+# setwd("/Users/zhaoxiang/Documents/tmp/EC_GLM_notebook") 
+
+workspace <- getwd()  # Get the current working directory and store it in 'workspace'
+
+# Increase the plot size by adjusting the options for plot dimensions in the notebook output
+options(repr.plot.width = 16, repr.plot.height = 8)  # Sets width to 16 and height to 8 for larger plots
+
+```
+
+## S.2 Install and load essential libraries.
+
+Install and load the required R packages. This step will automatically check for missing packages, install them if needed, and load them into the environment.
+
+```{r install_libraries, message=FALSE, warning=FALSE}
+
+# Set CRAN mirror
+options(repos = c(CRAN = "https://cran.rstudio.com/"))
+
+# List of packages to check, install if needed, and load
+packages <- c("spocc", "tidyverse", "leaflet", "sf", "rnaturalearth", "rnaturalearthdata")
+
+# Function to display a cat message
+cat_message <- function(pkg, message_type) {
+  if (message_type == "installed") {
+    cat(paste0(pkg, " has been installed successfully!\n"))
+  } else if (message_type == "loading") {
+    cat(paste0(pkg, " is already installed and has been loaded!\n"))
+  }
+}
+
+# Install missing packages and load them
+for (pkg in packages) {
+  if (!requireNamespace(pkg, quietly = TRUE)) {
+    install.packages(pkg)
+    cat_message(pkg, "installed")
+  } else {
+    cat_message(pkg, "loading")
+  }
+  library(pkg, character.only = TRUE)
+}
+
+library(spocc)
+library(tidyverse)
+library(leaflet)
+library(sf)
+library(rnaturalearth)
+library(rnaturalearthdata)
+```
+
+## Step 1 Overview and Conceptulisation : Notebook objective Taxon, location, predictors, scale
+
+# Introduction
+
+Shorebirds are key indicators of wetland health. This notebook demonstrates how to use `spocc` to fetch and visualise species occurrence data for:
+
+-   **Bar-tailed Godwit (*Limosa lapponica*)**
+-   **Red-necked Stint (*Calidris ruficollis*)**
+-   **Curlew Sandpiper (*Calidris ferruginea*)**
+
+These are long distance migratory shore birds which make their way to Australia and Tasmania, we will guide you how to check for species data through multiple available data collection.
+
+## Step 2 Get data: Listing our Bird species
+
+We use `spocc` to retrieve up to **500 occurrence records** per species from multiple data sources (GBIF, iNaturalist, OBIS).
```

Comment:
Please think if the bold statements are required. in here and across the other md sections of this notebook.